contracts/upgradeable_contracts/amb_erc677_to_erc677/HomeStakeTokenFeeManager.sol

```diff
@@ -61,4 +62,25 @@ contract HomeStakeTokenFeeManager is BlockRewardBridge, Ownable {
         uintStorage[FEE] = _fee;
         emit FeeUpdated(_fee);
     }
+
+    /**
+     * @dev Returns the state of the fee manager configuration: whether
+     * it is ready to collect and distribute fee or not.
+     */
+    function isFeeCollectingActivated() public view returns (bool) {
+        return ((address(_blockRewardContract()) != address(0)) && (getFee() > 0));
+    }
+
+    /**
+     * @dev Distributes fee as per the logic of the fee manager.
+     * In this particular case, the amount of fee is passed the block 
+     * reward contract which will mint new tokens and distribute them
+     * among the stakers.
+     * @param _fee amount of tokens to be distributed
+     */
+    function _distributeFee(uint256 _fee) internal {
+        if (address(_blockRewardContract()) != address(0)) {
```

Comment:
In `bridgeSpecificActionsOnTokenTransfer`, before calling `_distributeFee`, we call `isFeeCollectingActivated` that makes sure that block reward contract address is set. Should we also keep this other block reward check in case this method is called from different method?

Reply:
I just saw the same question was asked in another comment. I'm resolving this to continue the discussion there